nomad/structs/structs.go

```diff
@@ -7139,6 +7148,9 @@ const (
 
 // Allocation is used to allocate the placement of a task group to a node.
 type Allocation struct {
+	// msgpack omit empty fields during serialization
+	_struct bool `codec:",omitempty"` // nolint: structcheck
```

Comment:
Is there another way of doing this? Not a big deal, it would just be nice to avoid nolint comments when possible.

Perhaps we should do this globally on our msgpack encoder? I can't imagine why we'd ever want to serialize zero values.

Reply:
Unfortunately, this is the only way. I remember looking at it, and there's no global option for OmitEmpty.